### ISSUE TITLE
Image sent ok, but cannot download on phone. Popup message ask to change to another wifi?!

### ISSUE THREAD

**Issue by hgc2002** [AragurDEV/yowsup#47]
I've found some cases where the image is sent correctly from python (or that is how it looks) but it is not downloaded on whatsapp in the phone.

In the phone I see the typical blurred thumbnail to click and download the image (as it happens when there is no WiFi) but when I click to do it it says that cannot download and that I should change to another WiFi.

I have no log because python apparently has sent it ok. The issue is on the phone. Any idea how to deal with this?

Thanks.

**Comment by AragurDEV** [comment]
Thist seems like an decription error.
Try deleting your ~/.yowsup folder and test it again. (This will resync the keys)

**Comment by mikesalmonuk** [comment]
Hey @hgc2002 - do you notice if it's device specific? I've noticed it's a few iphone users who can't download the images, while non iphone users have no problem. The images are sent ok from the yowsup code, but seems an issue when it comes to downloading. Can't figure out where the problem is (i have cleared the ~/.yowsup folder a few times too)

**Comment by hgc2002** [comment]
I'll do it and let you know the results

**Comment by hgc2002** [comment]
folder found at /root and deleted. 
keys resynced. image test sent. 
failed same way.

**Comment by AragurDEV** [comment]
Thanks for all your help.
Seems it is an issue (again) that iOS users can't unencrypt images properly.

**Comment by mikesalmonuk** [comment]
So annoying when this happens. I wonder if Android will follow on a bit later with the same changes and be unable to decrypt the images?
Anyone with any suggestions know where to start looking at this to debug it?

**Comment by hgc2002** [comment]
I'll try android today and we will see how we are at this point in time

**Comment by mikesalmonuk** [comment]
How'd you get on @hgc2002? From what I've seen, android seems ok, and it's just (some) iOS devices that can't download the image. Sent ok from yowsup, but must be the way it's encoded or uploaded which caused iOS to fail to download on the client side.

**Comment by assegaf** [comment]
@mikesalmonuk how do you know its SOME iOS device only ? NOT all ios device ? ...
I just notice on ios latest 2.18.11 version and I also notice on status Tab, and recent update sample by Whatsapp can not be openned with same error wifi blah2 ..., the recent update also dont list any friend that android update. Bug by WhatsApp ?

**Comment by mikesalmonuk** [comment]
I have a group with about 15 people in it, and when yowsup sends an image, the android users can see it, and some iOS users can download it and the others get the wifi error message and cannot download it. I'll double check what versions of iOS and WhatsApp they're running, it might be older ones that still work.

**Comment by assegaf** [comment]
@mikesalmonuk if you didnt mind find out start from what version of whatsapp on ios that have error message popup. I can only confirm ios wa 2.18.11 have this trouble, the funny thing is the "Status Update Image  sample by Whatsapp " which show about the new whatsapp status image, can not be opened by ios wa 2.18.11 ,

**Comment by mikesalmonuk** [comment]
So it appears like it's working again for the iOS users. The few people who were having issues have started to see the pictures again today. No changes from my side, and they don't think or know if whatsapp auto updated, but seems to have fixed itself ok now. Anyone else notice it working again from today?

**Comment by assegaf** [comment]
do you think its a bug on whatsapp on  2.18.11 version ?  I see they uploaded 1.18.12 very fast, maybe a fix ... let me try, will post after it

**Comment by mikesalmonuk** [comment]
I think it may have been on their side with that version. I haven't had a chance to update anything, and everyone said it's now working again. So reckon a bug no their side with some encoded images.

**Comment by AragurDEV** [comment]
Okay so this bug is fixed in the latest version of WhatsApp for iOS?
Or are these special images that don't work?

**Comment by assegaf** [comment]
Yes confirmed, ios wa 2.8.11  Bug, now with update wa 2.18.12 its normal

**Comment by mikesalmonuk** [comment]
Bug seems to be fixed, which was on the client side anyway. So back to normal, nothing to do here with yowsup 👍

**Comment by AragurDEV** [comment]
Okay I can also confirm that it's working fine on Android 2.18.21 and iOS 2.18.12

**Comment by hgc2002** [comment]
yes, fixed in iOS 2.18.12 (already tested).
what a freak bug

**Comment by hgc2002** [comment]
Thanks for your help, buddies

**Comment by jegue** [comment]
on wa Android 2.18.24, the image can not be download from the app! image arrives ok! but got a download erro everytime!

**Comment by AragurDEV** [comment]
Yes same issue here

**Comment by assegaf** [comment]
2.18.24 is beta version. So not to be worried should be

**Comment by ahlermi** [comment]
On iOS 2.18.12 GIF is still a problem, it is not possible anymore to send GIF pictures with yowsup

**Comment by ahlermi** [comment]
http://www.unwetterzentrale.de/images/icons/regen-orange.gif

**Comment by jegue** [comment]
Videos too

**Comment by AragurDEV** [comment]
Edited the title to clarify.
If one media fails they may fail all

**Comment by AragurDEV** [comment]
I also made an interesting discovery.
On WhatsApp Web the decryption works fine even if the Android app cant download and decode the image.

**Comment by pabloruan0710** [comment]
I Have same issue here

**Comment by markus7017** [comment]
is anyone able to send audio, video, document with the AragurDEV fork?
I'm able to send image, but nothing else.

is anyone able to receive media files? I see the message, but the media file is still encrypted. I saw posts that people are able to receive media, but can't get it work.

**Comment by iskaira** [comment]
> is anyone able to send audio, video, document with the AragurDEV fork?
> I'm able to send image, but nothing else.
> 
> is anyone able to receive media files? I see the message, but the media file is still encrypted. I saw posts that people are able to receive media, but can't get it work.

same problem
